File: Python/대회/제2회 보라매컵 본선 Open Contest/전역_임무.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <deque>
#include <cstdlib>

using namespace std;

int main() {
    int n, m, p;
    cin >> n >> m >> p;
    vector<vector<int>> stage;

    for (int i = 0; i < n; i++) {
        vector<int> row;
        for (int j = 0; j < m; j++) {
            int item;
            cin >> item;
            row.push_back(item);
        }
        stage.push_back(row);
    }

    for (auto i : stage) {
        if (find(i.begin(), i.end(), -1) != i.end()) {
            int items = count(i.begin(), i.end(), -1);
            deque<int> current_stage(i.begin(), i.end());
            sort(current_stage.begin() + items, current_stage.end());

            for (auto j : current_stage) {
                if (p >= j) {
                    p += j;
                } else if (p < j && p * 2 > j && items > 0) {
                    items--;
                    p *= 2;
                    p += j;
                } else {
                    cout << 0 << endl;
                    exit(0);
                }
            }
        } else {
            for (auto j : i) {
                if (p >= j) {
                    p += j;
                } else {
                    cout << 0 << endl;
                    exit(0);
                }
            }
        }
    }

    cout << 1 << endl;
    return 0;
}

File: 본계정/대회/제2회 보라매컵 본선 Open Contest/전역_임무.py
import sys
from collections import deque
input = sys.stdin.readline

n,m,p = map(int,input().split())
stage = []

for i in range(n):
    stage.append(list(map(int,input().split())))

for i in stage:
    if -1 in i:
        items = i.count(-1)
        current_stage = deque(sorted(i)[items:])
        for j in list(current_stage):
            if p >= j:
                p += j
            elif p < j and p*2 > j and items > 0:
                items -= 1
                p *= 2
                p += j
            else:
                print(0)
                sys.exit(0)
    else:
        current_stage = deque(sorted(i)[items:])
        for j in list(current_stage):
            if p >= j:
                p += j
            else:
                print(0)
                sys.exit(0)

print(1)